ibis/spark/tests/test_api.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+import ibis
+
+pytestmark = pytest.mark.spark
+pytest.importorskip('pyspark')
+
+
+@pytest.fixture(scope='session')
+def expr():
+    schema = ibis.schema(
+        [
+            ('foo', 'int64'),
+            ('bar', 'string')
+        ]
+    )
+    t = ibis.table(
+        schema, name='tbl'
+    )
+    expr = t.projection(
+        ['bar', (t.foo + 1).name('new_foo')]
+    ).mutate(
+        bar_len=t.bar.length()
+    )
+    return expr
+
+
+def test_compile(client, expr):
+    assert client.compile(expr) == ibis.spark.compile(expr)
```

Comment:
Can you inline the SQL string here for comparison and test that both produce the same string? If the compiler is producing the empty string (for example) then this would still pass.